app/src/main/java/org/simple/clinic/instantsearch/InstantSearchEffectHandler.kt

```diff
@@ -73,6 +77,29 @@ class InstantSearchEffectHandler @AssistedInject constructor(
     }
   }
 
+  private fun checkIfPatientAlreadyHasAnExistingNHID(scheduler: Scheduler): ObservableTransformer<CheckIfPatientAlreadyHasAnExistingNHID, InstantSearchEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .observeOn(schedulers.computation())
+          .flatMap { patientRepository.patientProfile(it.patientId).subscribeOn(scheduler) }
```

Comment:
Why do we need to subscribe to patient profile changes here?

Reply:
Still trying to understand how to work with observables. I subscribed in order to read the data returned by the `patientRepository.patientProfile(it.patientId)` didn't know that the `.map` function on `.observeOn` takes care of it. Still learning RxJava better everyday.

Reply:
No, I mean there is no necessity to subscribe to changes at all here, right? We only need to query and check this once, for which `patientProfileImmediate` exists.

We should subscribe to change notifications only when we explicitly need to notified of changes to these records (for example, when displaying results in a list or something like that).